### PR TITLE
Use config gem to manage ArchiveSpace config

### PR DESCRIPTION
**Pull request by taylor-steve**

### CODE REVIEW

File: app/services/aspace_client.rb [hunk 1]
```diff
@@ -5,20 +5,29 @@
...
     @password = password
+
+    return unless uri.user
+
+    @user ||= uri.user
+    @password ||= uri.password
+    @base_url = uri.dup.tap do |u|
+      u.user = nil
+      u.password = nil
+    end.to_s
```
Comment (jcoyne) [inline app/services/aspace_client.rb:28]: This addition doesn't seem related to the config gem changes.

File: app/services/aspace_client.rb [hunk 2]
```diff
@@ -5,20 +5,29 @@
 
 # Client for making requests via the ArchivesSpace API
 class AspaceClient
-  attr_reader :url
+  attr_reader :base_url
 
-  def initialize(url: ENV.fetch('ASPACE_URL', nil),
+  # rubocop:disable Metrics/MethodLength
+  def initialize(url: Settings.aspace.url,
```
Comment (jcoyne) [inline app/services/aspace_client.rb:11]: Will this disrupt environments where `ASPACE_URL` is configured? https://github.com/search?q=repo%3Asul-dlss%2Fpuppet%20ASPACE_URL&type=code

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by jcoyne** [comment]
@taylor-steve is there a reason that this needed to be done?

File: config/initializers/config.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  # config.overwrite_arrays = true
+
+  # Defines current environment, affecting which settings file will be loaded.
+  # Default: `Rails.env`
+  #
+  # config.environment = ENV.fetch('ENVIRONMENT', :development)
+
+  # Load environment variables from the `ENV` object and override any settings defined in files.
+  #
+  # config.use_env = false
```
[the remark this answers is not in the document]
Comment (corylown) [reply config/initializers/config.rb:27]: We don't. `Settings.aspace.url` gets set to the value of `ENV['ASPACE_URL']` in settings.yml.

File: app/services/aspace_client.rb (hunk 2 above)
Comment (corylown) [reply app/services/aspace_client.rb:11]: It won't disrupt any environments where `ASPACE_URL` is set because the value of `Settings.aspace.url` is set to `ENV['ASPACE_URL']` in `settings.yml`

File: app/services/aspace_client.rb (hunk 1 above)
Comment (corylown) [reply app/services/aspace_client.rb:28]: This was added in order to set the ASpace credentials more easily in development as a one line local config in settings.local.yml:

```
aspace:
  url: http://user:password@whatever.edu
```

File: config/initializers/config.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  # Define ENV variable prefix deciding which variables to load into config.
+  #
+  # Reading variables from ENV is case-sensitive. If you define lowercase value below, ensure your ENV variables are
+  # prefixed in the same way.
+  #
+  # When not set it defaults to `config.const_name`.
+  #
+  config.env_prefix = 'SETTINGS'
```
Comment (jcoyne) [inline config/initializers/config.rb:36]: Why are we setting this if we aren't loading from the environment (line 27)?

Comment (corylown) [reply config/initializers/config.rb:36]: These are the default settings from the gem: https://github.com/rubyconfig/config/blob/master/lib/generators/config/templates/config.rb

**Review by jcoyne — APPROVED** [review]

**Comment by corylown** [comment]
@taylor-steve needs a rebase.
